src/JWT/JwtDecoder.cs

```diff
@@ -249,9 +248,9 @@ public void Validate(JwtParts jwt, params byte[][] keys)
             {
                 _jwtValidator.Validate(decodedPayload, asymmAlg, bytesToSign, decodedSignature);
             }
-            else
+            else if (algorithm is ISymmetricAlgorithm symmAlg)
             {
-                ValidSymmetricAlgorithm(keys, decodedPayload, algorithm, bytesToSign, decodedSignature);
+                _jwtValidator.Validate(keys, decodedPayload, symmAlg, bytesToSign, decodedSignature);

```

Comment:
Is the existing (non-algo specific) method `Validate()` still needed then?

Reply:
not sure...